src/link/Elf.zig

```diff
@@ -1711,7 +1711,7 @@ fn linkWithLLD(self: *Elf, arena: Allocator, tid: Zcu.PerThread.Id, prog_node: s
             .both => {}, // this is the default
         }
 
-        if (output_mode == .Exe) {
+        if (output_mode == .Exe or (output_mode == .Lib and link_mode == .dynamic)) {
```

Comment:
Just switch to using `is_exe_or_dyn_lib` instead.

Reply:
that simplifies it a lot, thanks